besu/src/main/java/org/hyperledger/besu/services/TraceServiceImpl.java

```diff
@@ -114,6 +117,9 @@ private void trace(final Block block, final OperationTracer tracer) {
                                 maybeParentHeader
                                     .flatMap(BlockHeader::getExcessDataGas)
                                     .orElse(DataGas.ZERO));
+
+                    tracer.traceStartTransaction(transaction.getGasLimit());
```

Comment:
Could we get the to/from addresses here?

Reply:
Would it be easier to pass in the Transaction object as the argument and let the tracer implementations pick out gas limit and to/from?

Reply:
Not only easier, but also probably smarter :)

Reply:
I did the change and now the parameter is a `Transaction` object. I needed to move the around the interface of the transaction, which causes a lot of changes in different files. But only the imports are changing, nothing else.